multiscene/js/portal.js

```diff
@@ -0,0 +1,60 @@
+import {Component, Property} from "@wonderlandengine/api";
+import { SwitchScene } from './switch';
+
+/**
+ * switch-scene
+ */
+export class Portal extends Component {
+	static TypeName = "portal";
+	/* Properties that are configurable in the editor */
+	static Properties = {
+		target: Property.object(),
+		translationDuration: Property.float(0.5),
+		initialTranslate: Property.float(30),
+		portalRotateSpeed: Property.float(1.0),
+	};
+
+	startBool = false;
+	startUpdate = true;
+
+	onActivate() {
+		this.target.translateObject([0, 0, this.initialTranslate]);
+	}
+
+	update(dt) {
+		this.object.rotateAxisAngleDegObject(
+			[0, 0, 1],
+			dt * 10000 * this.portalRotateSpeed,
+		);
+		this.playerPosition = this.target.getPositionWorld([]);
```

Comment:
No need to store `playerPosition` and `portalPosition` on `this` if it's used only in this method